arangod/Cluster/Maintenance.cpp

```diff
@@ -0,0 +1,799 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2018 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Kaveh Vahedipour
+/// @author Matthew Von-Maszewski
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Basics/StringUtils.h"
+#include "Basics/VelocyPackHelper.h"
+#include "Logger/Logger.h"
+#include "Cluster/FollowerInfo.h"
+#include "Cluster/Maintenance.h"
+#include "Utils/DatabaseGuard.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/Methods/Databases.h"
+
+#include <velocypack/Compare.h>
+#include <velocypack/Iterator.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+#include <algorithm>
+
+using namespace arangodb;
+using namespace arangodb::basics;
+using namespace arangodb::maintenance;
+using namespace arangodb::methods;
+
+static std::vector<std::string> cmp {
+  "journalSize", "waitForSync", "doCompact", "indexBuckets"};
+static std::string const CURRENT_COLLECTIONS("Current/Collections/");
+static std::string const CURRENT_DATABASES("Current/Databases/");
+static std::string const ERROR_MESSAGE("errorMessage");
+static std::string const ERROR_NUM("errorNum");
+static std::string const ERROR("error");
+static std::string const PLAN_ID("planId");
+static std::string const PRIMARY("primary");
+static std::string const SERVERS("servers");
+static std::string const SELECTIVITY_ESTIMATE("selectivityEstimate");
+static std::string const COLLECTIONS("Collections");
+static std::string const DB ("/_db/");
+static std::string const FOLLOWER_ID("followerId");
+static VPackValue const VP_DELETE("delete");
```

Comment:
this is not a valid VPack value. At least it will not be a VPack string. There is no way to create VPackSlices with fixed string values using the VPackSlice constructor, except encoding the VPack type byte as first byte of the string. A VPack String with 6 bytes payload has type byte `\0x46`, which must be prepended here.

Reply:
odd. it still works. what the hell?

Reply:
it's fine. just used to set value in `add`